### PR TITLE
fix(ingress-host-match-tls): repair policy and add tests

### PR DESCRIPTION
**Pull request by aslafy-z**
## Related Issue(s)

fixes https://github.com/kyverno/policies/issues/360

## Description

- Repair validation of the policy for multiple hosts.
- Add more tests.

## Checklist

- [x] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [x] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.

### CODE REVIEW

File: other/ingress_host_match_tls/ingress-host-match-tls.yaml [hunk 1]
```diff
@@ -39,7 +39,7 @@ spec:
           deny:
             conditions:
               all:
-              - key: "{{ request.object.spec.tls[].hosts[] }}"
+              - key: "{{ to_array(element.host) }}"
                 operator: AnyNotIn
```
Comment (aslafy-z) [inline other/ingress_host_match_tls/ingress-host-match-tls.yaml:43]:
```suggestion
                operator: AllNotIn
```

File: other/ingress_host_match_tls/ingress-host-match-tls.yaml [hunk 2]
```diff
@@ -42,4 +42,4 @@ spec:
               - key: "{{ request.object.spec.tls[].hosts[] }}"
                 operator: AnyNotIn
                 value:
-                - "{{ element.host }}"
\ No newline at end of file
+                - "{{ element.host }}"
```
[the remark this answers is not in the document]
Comment (Sissi44) [reply other/ingress_host_match_tls/ingress-host-match-tls.yaml:45]: > Another option would be:
> 
> ```yaml
>       deny:
>         conditions:
>           all:
>           - key: "{{sort(request.object.spec.rules[].host)}}"
>             operator: AllNotIn
>             value: "{{sort(request.object.spec.tls[].hosts[])}}"
> ```

Only this works for me ... AllNotIn operator not work for the multiple ingress use cases

File: other/ingress_host_match_tls/ingress-host-match-tls.yaml [hunk 3]
```diff
@@ -8,7 +8,7 @@ metadata:
     policies.kyverno.io/severity: medium
     kyverno.io/kyverno-version: 1.6.0
     policies.kyverno.io/minversion: 1.6.0
-    kyverno.io/kubernetes-version: "1.23"
+    kyverno.io/kubernetes-version: "1.20"
```
Comment (chipzoller) [inline other/ingress_host_match_tls/ingress-host-match-tls.yaml:11]: This indicates you tested and verified this policy using Kubernetes 1.20. Is that correct?

**Review by chipzoller — CHANGES_REQUESTED** [review]

File: other/ingress_host_match_tls/ingress-host-match-tls.yaml (hunk 3 above)
Comment (aslafy-z) [inline other/ingress_host_match_tls/ingress-host-match-tls.yaml:11]:
```suggestion
    kyverno.io/kubernetes-version: "1.20, 1.23"
```

File: other/ingress_host_match_tls/ingress-host-match-tls.yaml [hunk 4]
```diff
@@ -8,7 +8,7 @@ metadata:
     policies.kyverno.io/severity: medium
     kyverno.io/kyverno-version: 1.6.0
     policies.kyverno.io/minversion: 1.6.0
-    kyverno.io/kubernetes-version: "1.23"
+    kyverno.io/kubernetes-version: "1.20, 1.21, 1.23"
```
Comment (chipzoller) [inline other/ingress_host_match_tls/ingress-host-match-tls.yaml:11]: So you did in fact test this modified policy across all three of these versions? In the webhook? And it worked?

**Review by chipzoller — APPROVED** [review]

**Comment by chipzoller** [comment]
DCO sign-off is required.

File: other/ingress_host_match_tls/ingress-host-match-tls.yaml (hunk 4 above)
Comment (aslafy-z) [reply other/ingress_host_match_tls/ingress-host-match-tls.yaml:11]: I tested this modified policy on 1.20 and 1.21 only.

Comment (aslafy-z) [reply other/ingress_host_match_tls/ingress-host-match-tls.yaml:11]: Should I remove the mention for 1.23?

Comment (aslafy-z) [reply other/ingress_host_match_tls/ingress-host-match-tls.yaml]: I just removed the mention.

**Pull request merged**